External/SPMmodified/xASL/xASL_io_SaveNifti.m

```diff
@@ -60,18 +63,36 @@ function xASL_io_SaveNifti(pathOrigNifti, pathNewNifti, imNew, nBits, bGZip, cha
 	end
 end
 
+if nargin < 7 || isempty(bSaveJson)
+    bSaveJson = false;
+end
+
+if nargin < 8
+    JsonFields = struct;
+end
+
+
 % If the absolute path is missing and filename is given only, then add the current path to the absolute path
 % Do this both for the new filename and the original filename
-[pathstr, name0, ext0] = fileparts(pathNewNifti);
-if isempty(pathstr)
+[PathNew, NameNew, ExtNew] = xASL_fileparts(pathNewNifti);
+if isempty(PathNew)
 	% If a file-name only, then add the full current path to avoid ambiguity
-	pathNewNifti = fullfile(pwd(),[name0 ext0]);
+	pathNewNifti = fullfile(pwd(), [NameNew ExtNew]);
 end
 
-[pathstr, name0, ext0] = fileparts(pathOrigNifti);
-if isempty(pathstr)
+[PathOri, NameOri, ExtOri] = xASL_fileparts(pathOrigNifti);
+if isempty(PathOri)
+	% If a file-name only, then add the full current path to avoid ambiguity
+	pathOrigNifti = fullfile(pwd(), [NameOri ExtOri]);
+end
+
+pathOrigJson = fullfile(PathOri, [NameOri '.json']);
+pathNewJson = fullfile(PathOri, [NameNew '.json']);
+
+[PathOri, NameOri, ExtOri] = xASL_fileparts(pathOrigNifti);

```

Comment:
Is this duplicate by mistake? Lines 92-95 seem equivalent to 83-87

Reply:
Done